### DISCUSSION TITLE
`externals` load order

### DISCUSSION THREAD

**Discussion by imsys** [webpack/webpack#14452] [Q&A]
# Bug report

**What is the current behavior?**
It seems that external libraries are loading in different order each time.

**If the current behavior is a bug, please provide the steps to reproduce.**

```javascript
externalsType: 'script',
externals: {
  react: ['https://unpkg.com/react@17/umd/react.production.min.js', 'React'],
  'react-dom': [
    'https://unpkg.com/react-dom@17/umd/react-dom.production.min.js',
    'ReactDOM',
  ],
  'react-router-dom': [
    'https://unpkg.com/react-router-dom/umd/react-router-dom.min.js',
    'ReactRouterDOM',
  ],
  'react-is': [
    'https://unpkg.com/react-is@16.8.3/umd/react-is.production.min.js',
  "ReactIs"],
  'babel': ["https://unpkg.com/babel-standalone@6.26.0/babel.js", "Babel"],
}
```

Each time I refresh the page, sometimes it works, sometimes it doesn't. The most probable explanation I can think of is the load order.
Working:
![Selection_419](https://user-images.githubusercontent.com/911254/136674927-eb5727fc-c9ce-4b76-9c57-23651e272025.png)

Not working:
![Selection_418](https://user-images.githubusercontent.com/911254/136674921-dd1a005f-0e10-420e-925c-6d9921b6fcad.png)

Is there a way to force the load order? Or is just by writing <script> in the html file? I will use <script> in the meanwhile.

**What is the expected behavior?**
I expected to load in the order I specified.

**Other relevant information:**
webpack version:  5.58.1
Node.js version: 16.5.0
Operating System: Linux Manjaro, kernel 5.10.54-1
Additional tools: yarn, webpack-merge

**Comment by imsys** [comment]
You can get clone the `cdn-load2` branch of the repo I'm coding.
https://github.com/LionGovIO/LionGov/tree/cdn-load2

I think I only saw this behavior in production. So:
```
yarn
yarn build
yarn start
```

**Reply by alexander-akait** [reply]
We don't change order of externals, even more we don't change order, externals just load module from external place

**Reply by alexander-akait** [reply]
`externalsType: 'script'` uses `Promise`, but other deps uses `require` to load `React`, so it is not working, better to use default and include all scripts in HTML page

**Reply by imsys** [reply]
@alexander-akait The problem I'm seeing is that react has a script file for development and another for production, and if I want to use the CDN, I will need to have two index.html pages for each one. Or maybe during the build process I could include a script that would replace ".development.js" to ".production.min.js".
